### PR TITLE
Disregard OD variable description in SdoClient.upload()

### PR DESCRIPTION
**Pull request by acolomb**
The upload method should behave as a raw producer of byte data.
Interpretation of the returned data based on the Object Dictionary is
the responsibility of the `SdoVariable` class, which delegates to the
generic `ODVariable` decoding functions.

Move the data truncation to the method `SdoVariable.get_data()`, where
access to the `ODVariable` is certain.

The only truncation that still happens is based on the response size
specified by the server, which might be smaller for e.g. expedited
upload.  Extend the `upload()` function docstring to clarify the changed
behavior.  Adjust the test case for expedited upload with unspecified
size in the response, which is the only incompatible change.

This is based on, and includes the changes of, #591.

### CODE REVIEW

**Comment by acolomb** [comment]
Alright, this is ready for prime time with now complete test coverage.  Would still like to postpone it past the next release, as it significantly changes the return value of `SdoClient.upload()` in case an OD description is available.

File: canopen/sdo/client.py [hunk 1]
```diff
@@ -121,17 +125,8 @@ def upload(self, index: int, subindex: int) -> bytes:
...
-        if var is not None:
-            # Found a matching variable in OD
-            if var.fixed_size:
-                # Get the size in bytes for this variable
-                var_size = len(var) // 8
-                if response_size is None or var_size < response_size:
-                    # Truncate the data to specified size
-                    data = data[0:var_size]
+        if response_size and response_size < len(data):
+            data = data[:response_size]
```
[the remark this answers is not in the document]
Comment (sveinse) [reply canopen/sdo/client.py:129]: When making prototype one should hope for the best and plan for the worst and not just expect happy path. So I believe its just to figure out how to resolve when a remote server replies with odd responses.

There are datatypes which is variable, such as string (which happens to be used in the test for this PR), so the OD cannot resolve them all.

Many times I'd wished that the protocol test suite from CiA would be public so we could see how they resolve it.

**Review by sveinse — APPROVED** [review]
These changes looks good. There are probably more corner cases that can be pursued for SDO transfers, but that will have to be addressed in other PRs. Approved.
